iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,316 @@
+        {
+            string componentName = (string)userContext;
+            string propertyName = "targetTemperature";
+
+            (bool targetTempUpdateReceived, double targetTemperature) = PnpHelper.GetPropertyFromTwin<double>(desiredProperties, propertyName, componentName);
+            if (targetTempUpdateReceived)
+            {
+                _logger.LogDebug($"Property: Received - component=\"{componentName}\", {{ \"{propertyName}\": {targetTemperature}°C }}.");
+
+                string pendingPropertyPatch = PnpHelper.CreatePropertyEmbeddedValuePatch(
+                    propertyName,
+                    JsonConvert.SerializeObject(targetTemperature),
+                    ackCode: (int)StatusCode.InProgress,
+                    ackVersion: desiredProperties.Version,
+                    componentName: componentName);
+
+                var pendingReportedProperty = new TwinCollection(pendingPropertyPatch);
+                await _deviceClient.UpdateReportedPropertiesAsync(pendingReportedProperty);
+                _logger.LogDebug($"Property: Update - component=\"{componentName}\", {{\"{propertyName}\": {targetTemperature}°C }} is {StatusCode.InProgress}");
+
+                // Update Temperature in 2 steps
+                double step = (targetTemperature - _temperature[componentName]) / 2d;
+                for (int i = 1; i <= 2; i++)
+                {
+                    _temperature[componentName] = Math.Round(_temperature[componentName] + step, 1);
+                    await Task.Delay(6 * 1000);
+                }
+
+                string completedPropertyPatch = PnpHelper.CreatePropertyEmbeddedValuePatch(
+                    propertyName,
+                    JsonConvert.SerializeObject(_temperature[componentName]),
+                    ackCode: (int)StatusCode.Completed,
+                    ackVersion: desiredProperties.Version,
+                    serializedAckDescription: JsonConvert.SerializeObject("Successfully updated target temperature"),
+                    componentName: componentName);
+
+                var completedReportedProperty = new TwinCollection(completedPropertyPatch);
+                await _deviceClient.UpdateReportedPropertiesAsync(completedReportedProperty);
+                _logger.LogDebug($"Property: Update - component=\"{componentName}\", {{\"{propertyName}\": {_temperature[componentName]}°C }} is {StatusCode.Completed}");
+            }
+            else
+            {
+                _logger.LogDebug($"Property: Update - component=\"{componentName}\", received an update which is not associated with a valid property.");
+            }
+        }
+
+        // Send working set of device memory over telemetry.
+        private async Task SendDeviceMemoryAsync(CancellationToken cancellationToken)
+        {
+            string telemetryName = "workingSet";
+            long workingSet = Environment.WorkingSet / 1024;
```

Comment:
turns out workingset is not a great measure of memory usage as it includes shared memory used by other processes.

private bytes is better. https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.privatememorysize64?view=netcore-3.1